### ISSUE TITLE
Cannot Submit Solution to Catalog - No Error Message

### ISSUE THREAD

**Issue by inquisitive-identity** [microsoft/PowerPlatform-Catalog#27]
Hi team,

I am facing an issue with a freshly set up catalog on my tenant. The biggest problem is that the error message I get is not specific at all, but rather generic.

I am a PP Admin in the tenant and I installed the PP Catalog from AppSource, as instructed in the documentation.
I created an environment named _Power Platform Catalog Host_ where I'm planning on hosting all catalog items. I know it's redundant since I'm a System Admin in this environment, but I gave my account all Catalog-related security roles as well.

Then, I have 2 additional environments, one Sandbox, where I develop my solution, and the other Production, where I plan on consuming my solution. I am Sys Admin in both environments.

I set up everything properly (I hope) from the Catalog installation, to Publishers (just one for now), to Users within the Publishers (just my account for now), but when I try to submit a solution to the Catalog, I get the following error message.

![image](https://github.com/user-attachments/assets/1c210536-3690-4533-aefe-900e094d3bb7)

Do you know what I'm potentially doing wrong?

**App Info (please complete the following information):**
 - OS: Windows 10
 - Browser: MS Edge
 - Version: 126.0.2592.87 (Official build) (64-bit)

**Additional comment**

I was actually able to set up a Catalog successfully on my other demo tenant (which has no premium licenses whatsoever), but there I was another issue, which could be a separate topic.
Basically on the other tenant, I would successfully submit a solution through UI, approve it in the Catalog app, but the version remains the same - 1.0.0.0 - no matter what I do. As a result of this, My catalog never really receives the latest version and only keeps the initial one...

Thank you very much in advance!

Best regards,
Stefan

**Comment by MattB-msft** [comment]
Hi @inquisitive-identity, 
Sorry your having issues there..  we are working to improve the error reporting in the UX as part of the preview. 

A few questions on the submit via the UX.
on the first screen,  did you choose which catalog you wanted to submit too?  There are likely 2 of them.

Also, you can Set the name of the catalog that will appear in the Catalog Settings (Catalog Manager application -> Settings -> name ) 
![image](https://github.com/user-attachments/assets/4199bb85-6d67-473a-b5a0-9bce46d137d1)

Regarding versions, 
Version numbers are controlled by the solution export itself.  currently the catalog packaging leg is not incrementing the version on export from your development environment,  you will need to do that prior to submitting to the catalog. 

That said, it is replacing it on submission.  
On your catalog manager,  if you take a look at "Catalog Items",  you can change the view to "Active" from "Published" and you will see the other instances you had published previously

**Comment by inquisitive-identity** [comment]
Hi @MattB-msft 

Thank you very much for the swift response! :)

Yeah, I had already renamed the Default Solution to the name I wanted, and that's the only one I have in that tenant. 
It's such a strange error really because I cannot, for the life of me, figure out what I could have done wrong.
Do you think there could be some other tenant-level or some other setting that could be preventing me from submitting a solution, because somehow it does seem like it's tenant-specific.

On the other tenant it runs the submission fine, except for the versioning thing. 
So basically, I need to set the solution version in the solution settings before I submit it to the Catalog?

Best regards,
Stefan

**Comment by MattB-msft** [comment]
On the first, 
the name im talking about is the catalog name,  there are likly 2 catalogs in your tenant. 
for example in on of my test tenant: 
![image](https://github.com/user-attachments/assets/28186a6c-e0f2-44c5-9995-ddf092bd8478)

naming the catalog to something descriptive will make sure your sending your submission to the right place :) 

The next thing to do would be to use F12 and look at the browser traffic when you submit the submission request and read the 'real' error from there. 

on the versioning issue, Yes. just bump the version number in the solution before you submit to catalog.

**Comment by inquisitive-identity** [comment]
Yeah, I had renamed the Default Catalog - I only have one catalog in this tenant so I guess it is going to the right place. :) 
I will try to make another catalog in another environment to see if that will help.

Ah, I just tried the F12 but I apologize, I am not sure where to look as there are a lot of things there. :) 

The versioning thing worked flawlessly, thank you for that! :)

**Comment by MattB-msft** [comment]
Glad to hear the versioning worked for you there. 

Regarding F12. (getting the network error) 
Open the submission document and go through the process up to the last step. "Review + Finish"
Then use F12 to bring up the browser debugger. 
Then go to the Network tab. 
Then Click "Submit" on the window to submit the catalog item to the catalog. 
![image](https://github.com/user-attachments/assets/c50b8f11-9c06-4271-9a09-c512fd666f28)

You will see a lot of traffic go by and eventually you will see a "red" row with a status of "500"
![image](https://github.com/user-attachments/assets/87d5f78b-4788-47ce-9a02-7fa25732dec6)

Click that row in the debugger and a panel will open on the side with "Headers" chosen ,  Select the response tab and you should see a description of the error. 
If your ok with that data, please paste it here.

**Comment by inquisitive-identity** [comment]
Hi @MattB-msft,

Apologies for the delay - busy days...
Anyhow, I tried doing this, but the only errors I see in the log are these 3:
![image](https://github.com/user-attachments/assets/01101053-76bd-4840-a22b-649822b873b6)

And the description in the _Response_ area shows these, respectively:

![image](https://github.com/user-attachments/assets/2ef01ff6-45ee-4a69-9b8f-66bf026b6835)

![image](https://github.com/user-attachments/assets/f8741578-aaa9-4b9c-968c-66529271595f)

![image](https://github.com/user-attachments/assets/1cd775ea-ab54-4a2d-8aa5-b997ae00536d)

I'm not sure if these are helpful to you at all...
